### ISSUE TITLE
TypeError [ERR_INVALID_ARG_TYPE]: The "emitter" argument must be an instance of EventEmitter or EventTarget. Received an instance of AbortSignal

### ISSUE THREAD

**Issue by 345ml** [awinogrodzki/next-firebase-auth-edge#64]
Thanks, developer!

I get the error in the title.
I have confirmed that the environment variable is loaded.

```
next-firebase-auth-edge@^0.7.6:
  version "0.7.6"
  resolved "https://registry.yarnpkg.com/next-firebase-auth-edge/-/next-firebase-auth-edge-0.7.6.tgz#e7b1b8be27dfeed5f1bcd6720946b13ddfc34004"

next@^13.4.10:
  version "13.4.10"
  resolved "https://registry.yarnpkg.com/next/-/next-13.4.10.tgz#a5b50696759c61663d5a1dd726995fa0576a382e"
```

```ts
export const {
  getCustomIdAndRefreshTokens,
  verifyIdToken,
  createCustomToken,
  handleTokenRefresh,
  getUser,
  deleteUser,
  verifyAndRefreshExpiredIdToken,
  setCustomUserClaims,
} = getFirebaseAuth(
  {
    projectId: JSON.parse(process.env.NEXT_PUBLIC_FIREBASE_CONFIG!).projectId,
    privateKey: process.env.SERVICE_ACCOUNT_PRIVATE_KEY!.replace(/\\n/g, "\n"),
    clientEmail: process.env.SERVICE_ACCOUNT_CLIENT_EMAIL!,
  },
  JSON.parse(process.env.NEXT_PUBLIC_FIREBASE_CONFIG!).apiKey
);
```

```
- error TypeError [ERR_INVALID_ARG_TYPE]: The "emitter" argument must be an instance of EventEmitter or EventTarget. Received an instance of AbortSignal
    at new NodeError (node:internal/errors:399:5)
    at getEventListeners (node:events:914:9)
    at new Request (node:internal/deps/undici/undici:7138:17)
    at new NextRequest (/node_modules/next/dist/server/web/spec-extension/request.js:32:9)
    at NextRequestAdapter.fromNodeNextRequest (/node_modules/next/dist/server/web/spec-extension/adapters/next-request.js:65:16)
    at NextRequestAdapter.fromBaseNextRequest (/node_modules/next/dist/server/web/spec-extension/adapters/next-request.js:41:35)
    at RouteHandlerManager.handle (/node_modules/next/dist/server/future/route-handler-managers/route-handler-manager.js:22:57)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async doRender (/node_modules/next/dist/server/base-server.js:1006:38)
    at async cacheEntry.responseCache.get.incrementalCache.incrementalCache (/node_modules/next/dist/server/base-server.js:1215:28)
    at async /node_modules/next/dist/server/response-cache/index.js:99:36 {
  code: 'ERR_INVALID_ARG_TYPE'
}
```

**Comment by awinogrodzki** [comment]
Hey @345ml!

Looks like issue specific to Node.js. Could you share the versions of Node.js/NPM you're using?

**Comment by 345ml** [comment]
@awinogrodzki 
Thanks, response!

This is my version.

```
$ node -v
v18.15.0

$ npm -v
9.5.0
```

**Comment by awinogrodzki** [comment]
Hey @345ml !

I have reproduced the issue with Node.js 18.15. The problem was with outdated `@edge/ponyfill` library from Vercel. The issue will be fixed after replacing `@edge/polyfill` with `jose` library in https://github.com/awinogrodzki/next-firebase-auth-edge/pull/72

You can test it by installing pre-minor version `next-firebase-auth-edge@0.8.0-2`. Could you give it a try and let me know it worked? :)

**Comment by awinogrodzki** [comment]
Resolved in `next-firebase-auth-edge@0.8.0`

**Comment by 345ml** [comment]
@awinogrodzki 
Thanks!
Good job!!!
